index.js

```diff
@@ -1,39 +1,142 @@
+  },
+  searchRegWithout: {
+    expr: /\/search(~S\d*)?(\/([a-zA-Z]))?/,
+    handler: (match, query) => `${BASE_SCC_URL}/search?q=${getQueryFromParams(match[0], query)}${getIndexMapping(match[3])}`
+  },
+  patroninfoReg: {
+    expr: /\/patroninfo[^\/]\/(\d+)/,
+    handler: match => `${BASE_SCC_URL}?fakepatron=${match[1]}`
+  },
+  recordReg: {
+    expr: /\/record=(\w+)/,
+    handler: (match) => {
+      const bnum = match[1];
+      return `${BASE_SCC_URL}/bib/${bnum}`;
+    }
+  },
+};
+
+// The main method to build the redirectURL based on the incoming request
+// Given a path and a query, finds the first expression declared above which matches
+// the path, and returns the corresponding handler with the matchdata and query
+// As a default, returns the BASE_SCC_URL
+function mapWebPacUrlToSCCURL(path, query) {
+  console.log('mapping');
   let redirectURL;
-  const parsedURL = new URL(url);
-  const pathname = parsedURL.pathname
-  for (let pathType of expressions) {
-      const match = pathname.match(pathType.expr);
+  for (let pathType of Object.values(expressions)) {
+      const match = path.match(pathType.expr);
       if (match) {
-        redirectURL = pathType.handler(match, parsedURL);
+        console.log('matching: ', pathType);
+        redirectURL = pathType.handler(match, query);
         break
       }
   }
+  if (!redirectURL) redirectURL = BASE_SCC_URL;
+  console.log('returning: ', redirectURL);
   return redirectURL;
 }
 
 const handler = async (event, context, callback) => {
-  const redirectLocation = mapWebPacUrlToSCCURL(event.queryStringParameters.origin);
-  const response = redirectLocation ?
-    {
+  try {
+    console.log('env vars: ', process.env);
+    console.log('event: ', event.path, event.multiValueQueryStringParameters);
+    let path = event.path;
+    let query = event.multiValueQueryStringParameters;
+    let method = event.multiValueHeaders['x-forwarded-proto'][0] ;
+    let mappedUrl = mapWebPacUrlToSCCURL(path, query);
+    let redirectLocation = `${method}://${mappedUrl}`;
+    console.log('location: ', redirectLocation);
+    const response = {
+      isBase64Encoded: false,
       statusCode: 301,
-      headers: {
-        Location: redirectLocation,
-      }
-    }
-    : {
-      statusCode: 404
+      multiValueHeaders: {
+        Location: [redirectLocation],
+      },
+      body: JSON.stringify(console.log, null, 2)
     };
-  return callback(null, response);
+    return callback(null, response);
+  }
+  catch(err) {
+    console.log('err: ', err.message);
+    let method = event.multiValueHeaders['x-forwarded-proto'][0] ;
+    let mappedUrl = BASE_SCC_URL;
+    let redirectLocation = `${method}://${mappedUrl}`;
+    const response = {
+      isBase64Encoded: false,
+      statusCode: 301,
+      multiValueHeaders: {
+        Location: [redirectLocation],
+      },
+      body: JSON.stringify(console.log, null, 2)
```

Comment:
why stringify console.log?

Reply:
I saw this yesterday and I am also puzzled. I think this was just an accident and I am going to remove it, since we don't actually need a body